packages/forms/src/components/DateInput/DateInput.stories.tsx

```diff
@@ -0,0 +1,231 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { useWatch } from "react-hook-form";
+import { ComponentMeta, ComponentStory, Story } from "@storybook/react";
+import { action } from "@storybook/addon-actions";
+import { isBefore, parseISO } from "date-fns";
+
+import {
+  formatDate,
+  formDateStringToDate,
+  DATE_FORMAT_STRING,
+} from "@gc-digital-talent/date-helpers";
+import { PoolAdvertisement } from "@gc-digital-talent/graphql";
+import { Pending } from "@gc-digital-talent/ui";
+import { fakePoolAdvertisements } from "@gc-digital-talent/fake-data";
+
+import DateInput, { DateInputProps } from "./DateInput";
+import Form from "../BasicForm";
+import Submit from "../Submit";
+import { DATE_SEGMENT } from "./types";
+import Input from "../Input/Input";
+
+export default {
+  component: DateInput,
+  title: "Form/Date Input",
+  args: {
+    name: "date",
+    id: "date",
+    legend: "Date",
+  },
+  argTypes: {
+    show: {
+      control: {
+        type: "check",
+        labels: {
+          [DATE_SEGMENT.Year]: "Year",
+          [DATE_SEGMENT.Month]: "Month",
+          [DATE_SEGMENT.Day]: "Day",
+        },
+      },
+      options: [DATE_SEGMENT.Year, DATE_SEGMENT.Month, DATE_SEGMENT.Day],
+    },
+  },
+} as ComponentMeta<typeof DateInput>;
+
+type DateInputArgs = typeof DateInput;
+type DefaultValueDateInputArgs = DateInputArgs & {
+  defaultValue?: string;
+};
+
+const Template: ComponentStory<DefaultValueDateInputArgs> = (args) => {
+  const { defaultValue, ...rest } = args;
+  return (
+    <Form
+      {...(defaultValue && {
+        options: {
+          defaultValues: {
+            [rest.name]: defaultValue,
+          },
+        },
+      })}
+      onSubmit={(data) => action("Submit Form")(data)}
+    >
+      <DateInput {...rest} />
+      <Submit />
+    </Form>
+  );
+};
+
+export const Default = Template.bind({});
```

Comment:
Remember, "no yelling at the user until they've tried to do something wrong". :smile:  I'm not sure how you can "onBlur" when it's three inputs?  Maybe start it off at 0001-01-01 so they start off in a good state?

https://user-images.githubusercontent.com/8978655/225689299-29657848-5dfc-413e-ac61-76587234ccce.mp4

Reply:
Looks like the `BasicForm` uses `onChange` by default 🤔 Updated this story to use `onSubmit` in [update to on submit, fix conditional render story](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5928/commits/6638dd6b46366b817ea3d488aaab3801773e63bb)